internal/api/services/payment_initiations_retry.go

```diff
@@ -0,0 +1,96 @@
+package services
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/formancehq/go-libs/bun/bunpaginate"
+	"github.com/formancehq/payments/internal/models"
+	"github.com/formancehq/payments/internal/storage"
+	"github.com/pkg/errors"
+)
+
+func (s *Service) PaymentInitiationsRetry(ctx context.Context, id models.PaymentInitiationID) error {
+	adjustments, err := s.getAllPaymentInitiationAdjustments(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if len(adjustments) == 0 {
+		return errors.New("payment initiation's adjustments not found")
+	}
+
+	lastAdjustment := adjustments[0]
+
+	isReversed := false
+	switch lastAdjustment.Status {
+	case models.PAYMENT_INITIATION_ADJUSTMENT_STATUS_FAILED:
+	case models.PAYMENT_INITIATION_ADJUSTMENT_STATUS_REVERSE_FAILED:
+		isReversed = true
+	default:
+		return fmt.Errorf("cannot retry an already processed payment initiation: %w", ErrValidation)
+	}
+
+	pi, err := s.storage.PaymentInitiationsGet(ctx, id)
+	if err != nil {
+		return newStorageError(err, "cannot get payment initiation")
+	}
+
+	attempts := getAttemps(adjustments, isReversed)
+
+	if isReversed {
+		// TODO(polo): implement reverse
```

Comment:
If we're going to merge this with a TODO we should probably return an error here until the reversal is implemented